### ISSUE TITLE
Insufficient memcache settings

### ISSUE THREAD

**Issue by BjoernT** [rcbops/rpc-openstack#1048]
##### What version (commit) do you encounter this issue on?

All

##### What type of issue is this (bug, feature, documentation, regression)?

Feature

##### What is the expected behavior?

Current default settings are way to low in regards to memcache.
I recommend 8g RAM and >8k connection limit

##### What is actually happening?

1G RAM and 1K connections limits is hit quite often

##### How can we reproduce this issue?

N/A

##### How can we verify your issue is addressed?

Memcached uses the following opts

/usr/bin/memcached -m 8192 -p 11211 -u memcache  -c 2048 -t 4

##### If this is a large issue, what can be done immediately to alleviate the issue while working on a longer-term solution?

Minor
##### What is your evaluation on the importance of this issue, and why?

P3

###### Remove branches (lines) you do not wish to target.
- [ ] Master
- [ ] liberty-12.2
- [ ] liberty-12.1
- [ ] kilo

**Comment by BjoernT** [comment]
PR following

**Comment by git-harry** [comment]
I would like to see some more information, such as logs, regarding what is happening here. There is very little detail in this issue.

The issue is described as a feature yet it appears to be a bug.

You suggest that memcached_memory should be updated to 8192. Looking at https://github.com/openstack/openstack-ansible/blob/liberty/playbooks/roles/memcached_server/defaults/main.yml#L23 the current default is 8192 when there is at least 32768 on the box. So I would expect all customer deployments to already have memcached_memory set to 8192.

**Comment by mancdaz** [comment]
@BjoernT please address @git-harry comments before we can even consider the PR.

**Comment by BjoernT** [comment]
Yes I talked to @prometheanfire  this is on hold until I fix a requirement upstream.

**Comment by BjoernT** [comment]
Pending on https://review.openstack.org/#/c/317030/

**Comment by mancdaz** [comment]
Hi @bjoern I understand that, but the comments regarding how the issue has been reported still need to be addressed. Additionally, there is a question as to the validity of the problem, given what @git-harry pointed out.

**Comment by BjoernT** [comment]
Yes I get that but my issues are in kilo so none of that what git-harry said applies there,I have to back port some of that in a different way anyway. Also I wait until OSA handles nofile limits correctly otherwise extending the connections does not make any sense. So just wait until I got the OSA fixed to proceed here.

**Comment by mancdaz** [comment]
Hi @bjoern the osa review has now merged.  Could you please address the above comments regarding the bug report. Also, what @git-harry said about the default setting in the memcached playbook is true for kilo too:

https://github.com/openstack/openstack-ansible/blob/kilo/playbooks/roles/memcached_server/defaults/main.yml#L23

**Comment by Bjoern** [comment]
@mancdaz I think you mean @BjoernT - if you tag me, I receive a notification...

**Comment by BjoernT** [comment]
@mancdaz Yes I will update once I get it back ported to liberty

**Comment by mancdaz** [comment]
@BjoernT ping
